### PR TITLE
Move selectable functions to utils and convert to functional component.

### PR DESCRIPTION
**Pull request by dkasper-was-taken**
### Summary
Move selectable functions to utils, add refCallback, and convert to functional component.

Thanks for contributing to Terra. 
@cerner/terra

### CODE REVIEW

File: packages/terra-list/src/SelectableUtils.js [hunk 1]
```diff
@@ -86,6 +86,76 @@ const shouldHandleMultiSelect = (children, maxSelectionCount, currentIndexes, ne
  */
 const shouldHandleSingleSelect = (currentIndex, newIndex) => newIndex !== currentIndex;
 
+const wrappedOnClickForItem = (item, index, handleOnChange) => {
+  const initialOnClick = item.props.onClick;
+
+  return (event) => {
+    // The default isSelectable attribute is either undefined or true, unless the consumer specifies the item isSelectable attribute as false.
+    if (item.props.isSelectable !== false && handleOnChange) {
+      handleOnChange(event, index);
```
Comment (emilyrohrbough) [inline packages/terra-list/src/SelectableUtils.js]: IMO it'd be less confusing to just call `handleOnChange` `onChange` given this method is now outside of the selectable components and we directly pass the onChange provided to the selectableList.

File: packages/terra-list/src/SelectableUtils.js [hunk 2]
```diff
@@ -86,6 +86,76 @@ const shouldHandleMultiSelect = (children, maxSelectionCount, currentIndexes, ne
  */
 const shouldHandleSingleSelect = (currentIndex, newIndex) => newIndex !== currentIndex;
 
+const wrappedOnClickForItem = (item, index, handleOnChange) => {
```
Comment (emilyrohrbough) [inline packages/terra-list/src/SelectableUtils.js]: Maybe add some docs indicating what is these methods are doing.

File: packages/terra-list/src/SelectableUtils.js [hunk 3]
```diff
@@ -118,6 +124,9 @@ const wrappedOnKeyDownForItem = (item, index, handleOnChange) => {
   };
 };
 
+/**
+ * Returns an objet containing accessiblity and selectable projects.
```
Comment (emilyrohrbough) [inline packages/terra-list/src/SelectableUtils.js]: selectable ~projects~. properties?

Comment (dkasper-was-taken) [reply packages/terra-list/src/SelectableUtils.js]: good catch, i think the bbq is clouding my judgement

Comment (emilyrohrbough) [reply packages/terra-list/src/SelectableUtils.js]: Hey, its clouding mine too

**Pull request merged**